File: python/sound.py
#
# This file is part of elements.
# 
# Copyright (C) 2009-2011 William Oliveira de Lagos <william.lagos@icloud.com>
#
# Elements is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Elements is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with elements.  If not, see <http://www.gnu.org/licenses/>.
#

import platform

class Sound:
    def __init__(self):
        self.invaders = platform.load('audio')
        self.sounds = []
    def archive(self,file):
        self.sounds.append(invaders.load_sound(file))
    def play(self,namefile):
        for s in sounds:
            if namefile is s:
                invaders.play_sound(file)
    def stop_one(self,namefile):
        for s in sounds:
            if namefile is s:
                invaders.stop_sound(file)
    def stop(self):
        for s in sounds:
            invaders.stop_sound(file)

class Music:
    def __init__(self):
        pass
    
class Effects:
    def __init__(self):
        pass

File: src/sound.cc
/*
 * This file is part of elements project.
 * 
 * Copyright (C) 2009-2011 William Oliveira de Lagos <william.lagos@icloud.com>
 *
 * Elements is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Elements is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with elements.  If not, see <http://www.gnu.org/licenses/>.
 */

#include "CSound.h"

#define NUM_SOUNDS 2
struct sample {
    Uint8 *data;
    int dpos;
    int dlen;
} sounds[NUM_SOUNDS];

using namespace std;

CSound::CSound()
{

}

CSound::~CSound()
{

}

void CSound::PlaySound(char *wav)
{
    Mix_Chunk *music;
    Mix_OpenAudio(44100, AUDIO_S16SYS, 2, 2048);
    music = Mix_LoadWAV(wav);
    Mix_PlayChannel(1,music,0);
}

void CSound::PlayMP3(char *mp3){
    Mix_Music *music;
    Mix_OpenAudio(44100, AUDIO_S16SYS, 2, 2048);
    music = Mix_LoadMUS(mp3);
    Mix_VolumeMusic(100);
    Mix_PlayMusic(music,-1);
}
